### PR TITLE
Support for NodeJS v18

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

- Update dependencies to be compatible with Node 18.
- Update the CI config to run on Node 18.

#### Description of how you validated changes
Existing tests run successfully post upgrade.
https://app.circleci.com/pipelines/github/aws-amplify/amplify-codegen/2899/workflows/2f10290a-4b85-460e-9606-a8b85a9e3e70

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by manaswi223 — APPROVED** [review]

File: .circleci/config.base.yml [hunk 1]
```diff
@@ -103,8 +122,21 @@ jobs:
...
             - checkout
-            - run: nvm install 12.22.7
-            - run: nvm alias default 12.22.7
+            - run:
+                name: Install Node 18
+                command: |
+                  nvm version
+                  nvm install 18.15.0
+                  nvm use 18.15.0
+                  nvm ls
```
Comment (alharris-at) [inline .circleci/config.base.yml:131]: Is this here as a debug step?

File: .circleci/config.yml [hunk 2]
```diff
@@ -235,7 +269,9 @@ jobs:
           key: amplify-verdaccio-cache-{{ .Branch }}-{{ .Revision }}
       - node/install:
           install-yarn: true
-          node-version: '16.13'
+          node-version: '18.15'
+      - run: yarn config set workspaces-experimental true
+      - run: yarn cache clean --force
```
Comment (alharris-at) [inline .circleci/config.yml:274]: Why the change to workspaces and cache here?

File: .circleci/config.base.yml [hunk 3]
```diff
@@ -103,8 +122,21 @@ jobs:
...
           steps:
             - checkout
-            - run: nvm install 12.22.7
-            - run: nvm alias default 12.22.7
+            - run:
+                name: Install Node 18
+                command: |
+                  nvm version
+                  nvm install 18.15.0
+                  nvm use 18.15.0
```
[the remark this answers is not in the document]
Comment (dpilch) [reply .circleci/config.base.yml:130]: I added the `node/install` step for the android executor because it does not have `nvm` included. https://github.com/aws-amplify/amplify-codegen/blob/main/scripts/split-e2e-tests.ts#L135

I can't remember what happened when trying node/install on all the jobs, but it would be worth trying to possibly simplify the config.

[the remark this answers is not in the document]
Comment (phani-srikar) [reply .circleci/config.base.yml:130]: I've tried using the `node` orb for windows and it fails. Refer [this job](https://app.circleci.com/pipelines/github/aws-amplify/amplify-codegen/2904/workflows/b146be4a-750f-4272-a9cd-f972c9133496/jobs/16030) for error info.

[the remark this answers is not in the document]
Comment (phani-srikar) [reply .circleci/config.base.yml:130]:
```
$ nvm alias default 18.15.0
$ nvm use
```
is identical to `nvm use 18.15.0`. The later is concise.

File: .circleci/config.base.yml (hunk 1 above)
Comment (phani-srikar) [reply .circleci/config.base.yml:131]: Yes it is. It lists all the installed Node versions and highlights the **default**. We can get rid of it but I don't see any risk in keeping it.

File: .circleci/config.yml (hunk 2 above)
Comment (phani-srikar) [reply .circleci/config.yml:274]: The updated dependencies were not being reflected due to cache for the Android container. This causes the `build-app-android-test` job to fail. Refer [error logs in this failed run of the test](https://app.circleci.com/pipelines/github/aws-amplify/amplify-codegen/2887/workflows/63eec25c-45bb-4be0-b72e-8b74ab59bd92/jobs/15860).

**Review by alharris-at — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

**Review by dpilch — APPROVED** [review]
